com.unity.visualeffectgraph/Editor/Models/VFXModel.cs

```diff
@@ -327,7 +327,7 @@ private bool SetSettingValueAndReturnIfChanged(string name, object value)
             }
 
             var currentValue = setting.value;
-            if (currentValue != value)
```

Comment:
The variables are of type `object` which leads to reference comparison instead of value comparison.